tools/run-s390x-tests.sh

```diff
@@ -0,0 +1,111 @@
+#!/usr/bin/bash
+
+# This script prepares testing environment on a remote s390x
+# machine in beaker and runs specified tests on it over ssh.
+
+# Run on lastest nightly
+DISTRO=$(curl -L http://download.devel.redhat.com/rhel-8/nightly/RHEL-8/latest-RHEL-8.4/COMPOSE_ID)
+TEST_RUN=$1
+
+run_command () {
+    ssh root@$HOSTNAME $1
+}
+
+# Add beaker-client repo
+sudo cat > /etc/yum.repos.d/beaker-client.repo <<EOF
+[beaker-client]
+name=Baker Client - RHEL 8
+baseurl=http://download.eng.bos.redhat.com/beakerrepos/client/RedHatEnterpriseLinux8/
+enabled=1
+gpgcheck=0
+EOF
+
+# Install beaker-client
+sudo dnf install -y beaker-client
+
+# Recipe for beaker
+cat > beaker.xml <<EOF
+<job retention_tag="scratch">
+  <whiteboard>osbuild-composer s390x testing</whiteboard>
+  <recipeSet priority="High">
+    <recipe whiteboard="" role="RECIPE_MEMBERS" ks_meta="" kernel_options="" kernel_options_post="">
+      <autopick random="false"/>
+      <watchdog panic="ignore"/>
+      <packages/>
+      <ks_appends/>
+      <repos/>
+      <distroRequires>
+        <and>
+          <distro_family op="=" value="RedHatEnterpriseLinux8"/>
+          <distro_variant op="=" value="BaseOS"/>
+          <distro_name op="=" value="$DISTRO"/>
+          <distro_arch op="=" value="s390x"/>
+        </and>
+      </distroRequires>
+      <hostRequires>
+        <and>
+          <hostname op="like" value="s390x-kvm-%.lab.eng.rdu2.redhat.com"/>
+        </and>
+      </hostRequires>
+      <partitions/>
+      <task name="/distribution/check-install" role="STANDALONE"/>
+      <task name="/distribution/reservesys" role="STANDALONE">
+        <params>
+          <param name="RESERVETIME" value="7200"/>
+        </params>
+      </task>
+    </recipe>
+  </recipeSet>
+</job>
+EOF
+
+# Submit the job
+JOB_ID=$(bkr job-submit beaker.xml | awk -F'['\''|'\'']' '{print $2}')
+
+# Wait for the machine to get provisioned
```

Comment:
Can you walk me through how the provisioning/test+reporting of the results is intended to happen ? IMO we can eliminate the intermediary PSI machine (or at least keep it around for much shorter) depending on what's intended to happen.

Reply:
I guess you're right as this can easily be run from your local machine with the same results... Provisioning is as simple as submitting the above beaker.xml, which simply reserves a s390x machine that meets the requirements on line 47. That's a range of machines with the same hostname but different numbers, they are all the same and fit our needs. It periodically checks whether the machine is ready or gives up after 2 hours. Also it's reserving it only for 2 hours which should be more then enough time to run the tests. Then everything just happens over ssh, cloning the repository, building the rpms, installing them and then running the selected tests and returns the machine after all is done. It's expected to either just pass or exit with non zero. I guess I'm missing something like `set -e` but for that I would need better checking than what's on line 68 because that will fail the whole script otherwise.